bundler/spec/bundler/source/git/git_proxy_spec.rb

```diff
@@ -3,127 +3,177 @@
 RSpec.describe Bundler::Source::Git::GitProxy do
   let(:path) { Pathname("path") }
   let(:uri) { "https://github.com/rubygems/rubygems.git" }
-  let(:ref) { "HEAD" }
+  let(:ref) { nil }
+  let(:branch) { nil }
+  let(:tag) { nil }
+  let(:options) { { "ref" => ref, "branch" => branch, "tag" => tag }.compact }
   let(:revision) { nil }
   let(:git_source) { nil }
   let(:clone_result) { double(Process::Status, :success? => true) }
   let(:base_clone_args) { ["clone", "--bare", "--no-hardlinks", "--quiet", "--no-tags", "--depth", "1", "--single-branch"] }
-  subject { described_class.new(path, uri, ref, revision, git_source) }
```

Comment:
Note that the signature for GitProxy.new is not followed here, sending `'HEAD'` as `ref` instead of the actual expected argument of `options` as a `Hash`.  This is the case for all tests in this file, clearly indicating that these tests did zero testing of any `options`.  It was working only because of ducky handling:
```
>> opt = "HEAD"
=> "HEAD"
>> opt["tag"]
=> nil
```